meetings/working-groups/collection-literals/Core-interface-target-type-proposal.md

```diff
@@ -0,0 +1,247 @@
+
+Specifically:
+
+```c#
+void DoSomething(IEnumerable<int> values) { ... }
+
+// Not allowed.
+DoSomething([1, 2, 3]);
+
+// Write this instead:
+DoSomething((List<int>)[1, 2, 3]);
+```
+
+and
+
+```c#
+class Order
+{
+    // Not allowed:
+    public IReadOnlyList<OrderTag> Tags { get; } = [];
+
+    // Write this instead:
+    public IReadOnlyList<OrderTag> Tags { get; } = (ImmutableArray<OrderTag>)[];
+
+    // Which is worse than just:
+    public IReadOnlyList<OrderTag> Tags { get; } = Array.Empty<OrderTag>();
+}
+```
+
+First, this immediately fails the simplicity and universality concerns. From our own experience, and the data about the ecosystem, we know that users will need to interface with, well, interfaces :).  This will be an immediate hurdle that will deeply undercut the value of this feature space, potentially even deeply tainting it for the future.
+
+Second, collection expressions would not provide any sort of actual useful syntax or brevity for users in this case.  Because an explicit type would have to be provided, users would be left with syntax with nearly the same complexity and verbosity as what they would have to write today.  Users seeing this would rightfully ask "why would i pick this new form, that is really just the same as the old form, just with some tokens tweaked?"
+
+Performance and safety though would be mostly neutral here.  Users who did not care about either would likely just use `List<T>`, and users who did would pick something appropriate for their domain.  However, this would still be a small tick in the 'negative' category as our claims about us making the "right, smart, best choices" for users would be undercut by then forcing the user to have to make those choices themselves.
+
+### Option 2: Use `List<T>` for *all* of the possible interface targets.
+
+This is the next simplest option.  It says that for any of those five interface cases we are target-typing to, we should *always* use `List<T>` as the concrete type actually created.  Specifically:
+
+```c#
+void DoSomething(IEnumerable<int> values) { ... }
+
+// Legal. Is equivalent to `DoSomething(new List<int> { 1, 2, 3 })`
+DoSomething([1, 2, 3]);
+```
+
+and
+
+```c#
+class Order
+{
+    // Legal. Is equivalent to: `Tags { get; } = new List<OrderTag>();
+    public IReadOnlyList<OrderTag> Tags { get; } = [];
+}
+```
+
+This option *nails* the "simplicity", "universality, and "brevity" aspects we are trying to solve.  Explaining to users what happens in these cases is trivial: "It makes a `List<T>`.  The part-and-parcel type of .Net that you've know about and have used for nearly 20 years now".  Similarly, it can be used for *all* these apis that take in one of these interfaces.  Finally, it always allows the nice short syntactic-form that really sells people on why collection-expressions are a superior choice to use over practically every other collection-creation constructs they can use today.
```

Comment:
```suggestion
This option *nails* the "simplicity", "universality, and "brevity" aspects we are trying to solve.  Explaining to users what happens in these cases is trivial: "It makes a `List<T>`.  The part-and-parcel type of .NET that you've known about and have used for nearly 20 years now".  Similarly, it can be used for *all* these APIs that take in one of these interfaces.  Finally, it always allows the nice short syntactic form that really sells people on why collection expressions are a superior choice to use over practically every other collection creation construct they can use today.
```